src/Twig/Extension/CategoryFilterExtension.php

```diff
@@ -68,48 +55,30 @@ public function prepareData($currentValue, Category $rootCategory = null)
         return $data;
     }
 
-    /**
-     * @param int $currentValue
-     *
-     * @return Category
-     */
-    public function getCurrentCategory($currentValue)
+    public function getCurrentCategory(int $currentValue): Category
     {
-        return Category::getById($currentValue ?? -1);
+        return Category::getById($currentValue);
     }
 
     /**
-     * @param Category|null $currentCategory
-     * @param Category|null $rootCategory
-     * @return array|\Pimcore\Model\DataObject[]
+     * @return Category[]
      */
-    public function getSubCategories(Category $currentCategory = null, $rootCategory = null)
+    public function getSubCategories(Category $currentCategory = null, Category $rootCategory = null): array
     {
         $subCategories = [];
 
         $parent = $currentCategory ?: $rootCategory;
 
         if ($parent) {
-            $subCategories = array_filter($parent->getChildren(), function ($item) {
+            $subCategories = array_filter($parent->getChildren()->load(), static function ($item) {
                 return $item instanceof Category && $item->isPublished();
             });
-
-//            foreach($currentCategory->getChildren() as $subCategory) {
-//                $subCategories[] = $subCategory;
-//            }
         }
 
         return $subCategories;
     }
 
-    /**
-     * @param Category $category
-     * @param Category|null $rootCategory
-     * @param bool $reset
-     *
-     * @return string
-     */
-    public function generateLink(Category $category, Category $rootCategory = null, $reset = false): string
+    public function generateLink(Category $category, Category $rootCategory = null, bool $reset = false): string
```

Comment:
```suggestion
    public function generateLink(Category $category, ?Category $rootCategory = null, bool $reset = false): string
```

Reply:
Shouldn't we accept null also (like for eg: ?Category) for the above cases ?

Reply:
I added it to all nullable params.